Etapa1

NMDMaria opened the pull request.

NMDMaria committed:
.

NMDMaria committed:
.

NMDMaria committed:
.

NMDMaria committed:
trying to get the hang of services

NMDMaria committed:
rethinked.
-- formatting left

NMDMaria committed:
small description

NMDMaria committed:
added full description

NMDMaria committed:
added and fixed services
added multiple menus

NMDMaria committed:
Merge remote-tracking branch 'origin/Etapa1' into Etapa1

NMDMaria committed:
removed some extra classes

NMDMaria committed:
forgot an action

NMDMaria committed:
added and fixed services
added multiple menus

NMDMaria committed:
Merge remote-tracking branch 'origin/Etapa1' into Etapa1

NMDMaria committed:
fixed sort

NMDMaria committed:
moved mainMenu to main file from service

diana-maftei commented on .idea/uiDesigner.xml:
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[Fix] iti trebuie in proiect un .gitignore, in care sa pui tot ce tine de configurari intelij (.idea si fisierul .iml) si clasele compilate de java (pachetul out/)

diana-maftei commented on src/com/company/administrative/User.java:
```diff
@@ -0,0 +1,43 @@
+package com.company.administrative;
+
+public class User {
+    protected String username;
```
[Fix] astea sunt ok sa fie private, doar fiindca Doctor si Patient extind clasa asta, nu inseamna ca trebuie sa fie protected, si in niciintr-un caz sa nu fie accesate de catre servicii in mod direct, doar prin getteri si setteri

diana-maftei commented on src/com/company/administrative/UserService.java:
```diff
@@ -0,0 +1,465 @@
+        return null;
+    }
+
+    public Patient readPacient(Scanner scanner){
```
[Fix] asta ar trebui sa fie in PatientService

diana-maftei commented on src/com/company/administrative/UserService.java:
```diff
@@ -0,0 +1,465 @@
+        }
+    }
+
+    public Doctor readDoctor(Scanner scanner)
```
[Fix] asta ar trebui sa fie in DoctorService

diana-maftei commented on src/com/company/administrative/UserService.java:
```diff
@@ -0,0 +1,465 @@
+        }
+    }
+
+    private void doctorMenu(Doctor doctor, Scanner scanner)
```
[Fix] asta ar trebui sa fie in DoctorService

diana-maftei commented on src/com/company/administrative/UserService.java:
```diff
@@ -0,0 +1,465 @@
+        }while(!answer.equals("0"));
+}
+
+    private void patientMenu(Patient patient, Scanner scanner) {
```
[Fix] asta in patientService

diana-maftei commented on src/com/company/administrative/UserService.java:
```diff
@@ -0,0 +1,465 @@
+    }
+
+    /*
+    public void mainMenu(Scanner scanner)
```
[Improvement] nu lasa bucati de cod comentate in proiect, daca nu este folosita, sterge-o, daca vrei sa o folosesti in viitor, pune-o pe alt branch

diana-maftei commented on src/com/company/medicine/Affection.java:
```diff
@@ -0,0 +1,73 @@
+
+import java.time.LocalDate;
+
+public class Affection {
```
[Improvement] Affliction e cuvantul corect

diana-maftei commented on src/com/company/medicine/Affection.java:
```diff
@@ -0,0 +1,73 @@
+import java.time.LocalDate;
+
+public class Affection {
+    protected String name;
```
[Fix] campurile trebuie sa fie private

diana-maftei commented on src/com/company/medicine/Appointment.java:
```diff
@@ -0,0 +1,88 @@
+import java.util.Objects;
+
+public class Appointment {
+    protected LocalDateTime date;
```
[Fix] campuri private

diana-maftei commented on src/com/company/medicine/Appointment.java:
```diff
@@ -0,0 +1,88 @@
+        this.medicalProcedure = medicalProcedure;
+    }
+
+    public Appointment(Integer day, Integer month, Integer year,Integer hour, Integer minute, Status status, String specialization, MedicalProcedure medicalProcedure) {
```
[Fix] poti in constructorul asta sa apelezi constructorul de mai sus, fiindca seteaza aceleasi campuri, nu e nevoie sa duplici logica

diana-maftei commented on src/com/company/medicine/Checkup.java:
```diff
@@ -0,0 +1,73 @@
+import java.util.List;
+
+public class Checkup extends MedicalProcedure {
+    protected Affection diagnosis;
```
[Fix] campuri private

diana-maftei commented on src/com/company/medicine/Checkup.java:
```diff
@@ -0,0 +1,73 @@
+    protected String motive;
+    protected String observations;
+
+    public Checkup(LocalTime start_time, String motive) {
```
[Fix] fara _ in nume de parametri, iar in nume de campuri doar daca sunt contante (static, final) si alea se scriu cu litere mari

diana-maftei commented on src/com/company/medicine/Doctor.java:
```diff
@@ -0,0 +1,133 @@
+import java.util.Objects;
+
+public class Doctor extends User {
+    protected String firstName;
```
[Fix] campuri private

diana-maftei commented on src/com/company/medicine/DoctorService.java:
```diff
@@ -0,0 +1,230 @@
+            appointment.medicalProcedure.setStartTime(LocalTime.of(hours, minutes));
+
+
+            if (appointment.medicalProcedure instanceof Checkup) {
```
[Improvement] if-ul asta se poate extrage intr-o metoda privata handleCheckup()

diana-maftei commented on src/com/company/medicine/DoctorService.java:
```diff
@@ -0,0 +1,230 @@
+                }
+
+                appointment.medicalProcedure.setDuration(LocalTime.of(hours, minutes));
+            } else if (appointment.medicalProcedure instanceof Surgery) {
```
[Improvement] if-ul asta se poate extrage intr-o metoda privata handleSurgery()

diana-maftei commented on src/com/company/medicine/Patient.java:
```diff
@@ -0,0 +1,98 @@
+import java.util.Objects;
+
+public class Patient extends User {
+    protected String firstName;
```
[Fix] campuri private

diana-maftei commented on src/com/company/medicine/Treatment.java:
```diff
@@ -0,0 +1,42 @@
+package com.company.medicine;
+
+public class Treatment {
+    protected String drug;
```
[Fix] campuri private

diana-maftei commented on src/com/company/medicine/Doctor.java:
```diff
@@ -0,0 +1,133 @@
+package com.company.medicine;
```
[Fix] clasele din acest pachet nu sunt separate nici pe layere, nici dupa business logic, sunt cam la gramada

diana-maftei commented on src/com/company/medicine/PatientService.java:
```diff
@@ -0,0 +1,160 @@
+        else throw new Exception("No appointment at given date.");
+    }
+
+    public void moveAppointment(Patient patient, LocalDateTime date, LocalDateTime newDate) throws Exception
```
![image](https://user-images.githubusercontent.com/67580205/161293899-1ff42c8d-a228-457e-a8a7-ffca5f258dc4.png)
[Fix] cand am incercat sa move appoinment imi cerea un format de timp care mi-a zis ca e invalid

diana-maftei replied:
[Improvement] also, meniul cred ca ar fi fost mult mai usor de navigat daca listai optiunile de appointments si le dadeai un index dupa care sa le aleaga userul, decat sa bage mereu de mana data si ora

diana-maftei reviewed (commented):
Fain proiect, mi-a placut! Good job! Ti-am lasat niste comentarii cu ce poti imbunatati.

Ce e cu [Improvement] e doar sugestie de clean code, nu o chestie pe care am punctat-o din cerinte. Cele cu [Fix] sunt cele pe care trebuie sa le adresezi fiindca nu se conformeaza cerintelor.

NMDMaria committed:
fixed after code review

NMDMaria committed:
fixed after code review

NMDMaria committed:
changed .gitignore

NMDMaria committed:
Delete out directory

NMDMaria committed:
Delete .idea directory

NMDMaria committed:
change from interface back to class
not sure if they should have been interfaces or not, or if implemented them ok as interfaces
so back to public class with static methods and no fields...

NMDMaria committed:
Merge remote-tracking branch 'origin/Etapa1' into Etapa1

diana-maftei reviewed (approved):
Poti sa faci merge
